src/Stats.AggregateCdnDownloadsInGallery/AggregateCdnDownloadsJob.cs

```diff
@@ -61,45 +72,39 @@ public class AggregateCdnDownloadsJob : JsonConfigurationJob
 
         private AggregateCdnDownloadsConfiguration _configuration;
         private int _commandTimeoutSeconds;
+        private HttpClient _httpClient;
 
         public override void Init(IServiceContainer serviceContainer, IDictionary<string, string> jobArgsDictionary)
         {
             base.Init(serviceContainer, jobArgsDictionary);
 
             _configuration = _serviceProvider.GetRequiredService<IOptionsSnapshot<AggregateCdnDownloadsConfiguration>>().Value;
             _commandTimeoutSeconds = _configuration.CommandTimeoutSeconds ?? _defaultCommandTimeoutSeconds;
+            _httpClient = new HttpClient();
         }
 
         public override async Task Run()
         {
-            // Gather download counts data from statistics warehouse
-            IReadOnlyList<DownloadCountData> downloadData;
-            Logger.LogInformation("Using batch size {BatchSize} and batch sleep seconds {BatchSleepSeconds}.",
-                _configuration.BatchSize,
-                _configuration.BatchSleepSeconds);
-
-            var stopwatch = Stopwatch.StartNew();
+            var featureFlagRefresher = _serviceProvider.GetRequiredService<IFeatureFlagRefresher>();
+            await featureFlagRefresher.StartIfConfiguredAsync();
 
-            using (var connection = await OpenSqlConnectionAsync<StatisticsDbConfiguration>())
-            using (var transaction = connection.BeginTransaction(IsolationLevel.Snapshot))
+            IReadOnlyCollection<DownloadCountData> downloadData = null;
+            var jsonConfigurationAccessor = _serviceProvider.GetService<IOptionsSnapshot<DownloadsV1JsonConfiguration>>();
+            if (jsonConfigurationAccessor == null || jsonConfigurationAccessor.Value == null)
             {
-                Logger.LogInformation("Gathering Download Counts from {DataSource}/{InitialCatalog}...", connection.DataSource, connection.Database);
-
-                downloadData = (
-                    await connection.QueryWithRetryAsync<DownloadCountData>(
-                        _storedProcedureName,
-                        transaction: transaction,
-                        commandType: CommandType.StoredProcedure,
-                        commandTimeout: TimeSpan.FromSeconds(_commandTimeoutSeconds),
-                        maxRetries: 3))
-                    .ToList();
+                downloadData = await GetDownloadDataFromStatsDbAsync();
             }
+            else
+            {
+                var ff = _serviceProvider.GetRequiredService<IFeatureFlagClient>();
+                string url = jsonConfigurationAccessor.Value.SqlPipelineUrl;
+                if (ff.IsEnabled(AlternateStatisticsSourceFeatureFlagName, false))
```

Comment:
Please add named param:

```suggestion
                if (ff.IsEnabled(AlternateStatisticsSourceFeatureFlagName, defaultValue: false))
```